kafka-client/src/commonMain/kotlin/com/icemachined/kafka/common/serialization/JsonDeserializer.kt

```diff
@@ -0,0 +1,29 @@
+package com.icemachined.kafka.common.serialization
+
+import com.icemachined.kafka.clients.consumer.Headers
+import com.icemachined.kafka.common.header.KafkaHeaders
+
+import kotlin.reflect.KType
+import kotlinx.cinterop.toKString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.serializer
+
+/**
+ * JsonDeserializer
+ */
+class JsonDeserializer<T>(
```

Comment:
why do you need this wrapper?

Reply:
this incapsulate some logic by type extraction and serialization through kafka headers.
also it can be pre configured based on kafka konfiguration.
this is based on standard interface from kafka api

Reply:
so you expect these classes to be overriden by users?

Reply:
this class i don't think it should be overriden. never met this in spring.
may be we add some features to it in future, like setting fixed class type for serialize deserialize,
there can be cases there is no information in header about type.
also we can parametrize type extraction strategy for user.